usr/src/uts/common/fs/zfs/sys/dmu.h

```diff
@@ -564,12 +564,7 @@ typedef struct dmu_buf_user {
  * NOTE: This function should only be called once on a given dmu_buf_user_t.
  *       To allow enforcement of this, dbu must already be zeroed on entry.
  */
-#ifdef __lint
-/* Very ugly, but it beats issuing suppression directives in many Makefiles. */
```

Comment:
gcc will detect unused statics correctly but not get tripped up by inline functions. So it's true that lint won't notice, but as long a we aren't gagging unused functions, it should be fine.